app/src/main/java/org/p2p/wallet/auth/web3authsdk/mapper/Web3AuthClientMapper.kt

```diff
@@ -1,30 +1,39 @@
 package org.p2p.wallet.auth.web3authsdk.mapper
 
 import com.google.gson.Gson
+import org.p2p.wallet.auth.web3authsdk.Web3AuthApi.Web3AuthSdkInternalError
 import org.p2p.wallet.auth.web3authsdk.response.Web3AuthErrorResponse
 import org.p2p.wallet.auth.web3authsdk.response.Web3AuthSignInResponse
 import org.p2p.wallet.auth.web3authsdk.response.Web3AuthSignUpResponse
 import org.p2p.wallet.utils.fromJsonReified
 import timber.log.Timber
 
+private const val INTERNAL_ERROR_MESSAGE = "Web3Auth mapping failed"
+
 class Web3AuthClientMapper(private val gson: Gson) {
-    fun fromNetworkSignUp(responseJson: String): Web3AuthSignUpResponse? {
-        return kotlin.runCatching { gson.fromJsonReified<Web3AuthSignUpResponse>(responseJson) }
-            .onFailure { Timber.i(it) }
-            .getOrNull()
+    fun fromNetworkSignUp(responseJson: String): Result<Web3AuthSignUpResponse> = try {
+        Result.success(gson.fromJsonReified<Web3AuthSignUpResponse>(responseJson)!!)
+    } catch (mappingError: Throwable) {
+        Timber.i(mappingError)
+        Timber.i(responseJson)
+        Result.failure(Web3AuthSdkInternalError(INTERNAL_ERROR_MESSAGE, mappingError))
     }
 
-    fun fromNetworkSignIn(responseJson: String): Web3AuthSignInResponse? {
-        return kotlin.runCatching { gson.fromJsonReified<Web3AuthSignInResponse>(responseJson) }
-            .onFailure { Timber.i(it) }
-            .getOrNull()
+    fun fromNetworkSignIn(responseJson: String): Result<Web3AuthSignInResponse> = try {
+        Result.success(gson.fromJsonReified<Web3AuthSignInResponse>(responseJson)!!)
+    } catch (mappingError: Throwable) {
+        Timber.i(mappingError)
+        Timber.i(responseJson)
```

Comment:
it would be wise to log broken JSON what you think?

Reply:
but can be logged in same line

Reply:
im not sure how :(